ripple/src/main/webapp/businessdetails.js

```diff
@@ -106,6 +160,123 @@ function reviewButtonDisplay() {
   }
 }
 
+
+// Listen for realtime updates in user reviews and display each review
+function loadUserReviews() {
+  console.log("loadUserReview()");
+  var listenToReviews = (querySnapshot) => {
+    // Get outer element
+    var userReviewsElement = document.getElementById("user-reviews");
+    querySnapshot.forEach((doc) => {
+      // Read review data
+      var starRating = doc.data().starRating;
+      var priceRating = doc.data().priceRating;
+      var reviewText = doc.data().reviewText;
+      var numLikes = doc.data().numLikes;
+      var numDislikes = doc.data().numDislikes;
+      // Convert timestamp to a formatted date
+      var timestamp = doc.data().timestamp;
+      var date = timestamp.toDate().toDateString();
+      var formattedDate = parseDate(date);
+      var reviewUserId = doc.id;
+      console.log("doc: ",doc.id, starRating, priceRating, 
+          reviewText, formattedDate, numLikes, numDislikes);
+      // Use reviewUserId to get the name and blobKey of the user who commented
+      var getReviewUserInfo = (doc) => {
+        if (doc.exists) {
+          var userName = doc.data().userName;
+          var userBlobKey = doc.data().userBlobKey;
+          // Create review element
+          userReviewsElement.appendChild(createReviewElement(doc.id, userName, userBlobKey, starRating, priceRating, 
+              reviewText, formattedDate, numLikes, numDislikes));
+        } else {
+          console.log("Document does not exist");
+        }          
+      }
+      getDocByDocId("users", reviewUserId, getReviewUserInfo);
+    })
+  }
+  getOrSnapshotDocsByQuery("snapshot", "reviews", listenToReviews, [], [], "timestamp", "desc");
```

Comment:
nit: if a parameter can take in something empty, I think its good practice setting the default param to `[]` so you don't have to specify

Reply:
This it the current function: getOrSnapshotDocsByQuery(method, collection, lambda, whereFieldArray, whereValueArray, orderByField=false, orderDirection=false, varArray=false)

If I don't specify the empty array, I'm not sure how else the function will recognize the timestamp as the orderDirection

Reply:
I believe Javascript supports named parameters, so that shouldn't be an issue, but not a blocker

Reply:
Got it, will keep in mind for future